### ISSUE TITLE
Animation Library

### ISSUE THREAD

**Issue by Discord-Github-Bridge** [Yarden-zamir/Create-Prepare-to-Dye#496]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194730593846558810) 
> Tulkin wanted to make an animation library - but as he is busy it has fallen on us to pick up his burdens 💪 We must not disappoint

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194732414015131800) 
> **What are we animating?**
We're animating `create:contraption`s by setting their nbt to a certain premade-structures 

**How are is the nbt data for the structures created?**
Right now they're hard coded. Could we put these into a file and read them. (Has to be a json; or we add a read function to ptdye-plus)

**What is animated currently? What does it do?**
Right now; the only thing it really does is handle the "opening" sequence of the trading platform. Which requires it increasing by Y amount every tick until it reaches the desired height where it is then made solid from the `create:contraption`

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733358341697556) 
> # What is needed?
We need a way to smi-quickly define structures and their animations and be able to call them through lets say a minecraft command. 

Basically we need three main things
- a way define structure
  - a way to inject some data, like we do for the starter trades in the first trading platform 
- a way to spawn structure
- a way to assign animation sequence to structutre that gets triggered either on spawn or potentioally later on (not critical, on spawn is good enough)

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733665968738345) 
> How bad is generating json from nbt or vice versa?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733755848478881) 
> If we can hold the structure data in an object at runtime that will be the best

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733786156519557) 
> Well we can just have a single string field

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733834462298112) 
> Well but I want to manipulate it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733851877048320) 
> So that's not that good

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733875667144724) 
> mhm

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733954314539118) 
> I think making nbt from json could work, kjs has some helper functions that we can highjack

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194733992092651670) 
> But making json from nbt...

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194734082333098014) 
> Which we need if we want to build something and use it instead json define each block

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194734320527626311) 
> So define each block as an json object?

```json
[
  { "blocka": "dirt", "nbt":"{\t"I_am\":\"dirt\"}" },
  { "blocka": "container", "nbt":"{\"insertdata\":\"${here}\"}"
]
```

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194734822048931972) 
> I think the biggest issue is creating these structures in a readable-or even an editable format

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194734896107769909) 
> Since; in nbt world the position is a hash of the 3 axises

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194735484086263851) 
> Oh; there's `NBT.toJson()` function in kjs globals. No idea what it actually does

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194736612710555758) 
> https://github.com/KubeJS-Mods/Rhino/blob/1.19/main/common/src/main/java/dev/latvian/mods/rhino/mod/util/NBTUtils.java#L403

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194737252782329937) 
> *But there's no `fromJson` function*

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194737638826062014) 
> Ah; there's `fromTag` which has an edge case for the json format

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1194773308122595519) 
> Update time (mostly for self-reflection)
I found how to find glue. Which doesn't sound exciting... BUT with this new found power instead of making nbts with limited knowledge.
You can make a build; select it with glue and BAM we can find it in code to convert it into all the things we need to do
https://github.com/EngineerSmith/Create-Prepare-to-Dye/blob/animation/kubejs/server_scripts/base/animation/commands.js

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195140996409864283) 
> Update time
Been busy (and got distracted with 'art' so now there's a new painting in the game). 

Made a block that can be placed next to 'glued' structures. So that when you right click it, it saves the time of having to type any commands as it can all be done with a click. It will then highlight (similarly to glue) the connected blocks of the structure so the builder can check that everything is correctly included.

Current issue is that I found `BlockEvents.rightClicked` is being triggered twice in a row. Not sure what is causing it as it seems they're both running server side; so it's not running once on client and then on server... I'll have to keep looking into it.

Anyway; once this is done. I think adding some sort of interface (<:cry:1184476833090113606> ) for naming the structure. Then some sort of item, which you can link structures together and start animating them (without affecting the already built ones - think ghost blocks of create's schematics)

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195283614775124038) 
> Twice one for each hand!

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195283664565714985) 
> Check if main hand and it should work

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195283868085932044) 
> I seems like you are making it player facing, that sounds like a lot more work

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195302263451820042) 
> Just trying to make it UX friendly than end user friendly

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195302878152245278) 
> Also

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195302956866740256) 
> The end results will be "two" systems which work together

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195303190082629652) 
> 1st the creation of the animation; which will produce a json of keyframes (with functions like lerps to positions, and actions like adding/removing blocks (e.g. upside down fire))
2nd reading the jsons and putting them into action when it's called for it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195324126718595083) 
> Do you know how to add block entity values to jade? So they show when you look at the block?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195325812153528331) 
> not sure but [kikis2111](https://discord.com/channels/890222432605057044/1194730593846558810/1195325812153528331) did something similar with the ponder tooltip thing maybe he knows

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195376107193630790) 
> [yarden.zamir](https://discord.com/channels/890222432605057044/1194730593846558810/1195376107193630790)What's the mod that allows you to shift+click to pick up a block entity? And do you know how I can blacklist my block? If you don't know I can look into it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195383755087564870) 
> You know your luck for a simple solution is out when you find [autistic.sack.of.friends](https://discord.com/channels/890222432605057044/1194730593846558810/1195383755087564870) asking a similar question https://discord.com/channels/303440391124942858/1149285329417797672/1149285329417797672

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/145122297257132032/9b7bfe09029ad9f0e8dc639015ae8838.webp" width="15" height="15" center=true/> **kikis2111** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384578148401263) 
> Yeah its possible with the jade api if you add on to the java mod

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384665721286796) 
> json serialization isn't so bad, but not terribly efficient

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/145122297257132032/9b7bfe09029ad9f0e8dc639015ae8838.webp" width="15" height="15" center=true/> **kikis2111** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384667944267776) 
> Code in these files is related: https://github.com/jasperalani/ptdye-plus/tree/main/src/main/java/com/kikis/ptdyeplus/integration

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384693881847908) 
> i don't know if it works for blocks

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384710520656084) 
> probably not...

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384788878622870) 
> I need the block position *every* frame

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195384997264236604) 
> can't you just store a time and an animation id

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385060761808937) 
> and an origin

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385124091592775) 
> This is for the blocks within the structure

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385197202518127) 
> OHH big brain moment

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385322742239333) 
> isn't that a default feature of create structures...

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385336596004977) 
> I can just send the client aabb information and then let the client find the blocks instead of storing the blocks in nbt/entity data

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385564636139580) 
> While it won't be the most accurate; it should work fine as it'd use the same create function as the server is doing

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385617748594738) 
> and order doesn't matter for this

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/401254336136478732/32f2db537b37df8a123494331b0de45e.webp" width="15" height="15" center=true/> **autistic.sack.of.friends** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195385924620648509) 
> i meant if the bounding box thing runs after the blocks are placed

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195386193492332677) 
> Imagine create glue visual outline; do you notice it lagging as it only draws 1/20 a second than the frame rate

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195386322328764427) 
> Unless you mean for the actual animation

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195386411025715200) 
> What I'm doing is just to put a box around the structure so the builder knows what is included in the structure

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195386457796378664) 
> Although - I don't really need to do that now that I think about it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195386579863228436) 
> > Yeah; so just press ctrl while holding your super glue to make sure it is all connected

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195386648779817051) 
> 🧠  text > hours of work

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1195464346499817533) 
> Update time (As always for self reflection)

I've seen more of a vision forward for how I want the workflow to go now. Optionally, you will be able to use my "animation block" or anything else that will save NBT structures. The benfit of using the animation block is that you will see the boundary as you build onto the structure. The builder will be required to place a renamed name tag inside any container or inventory inside the structure so it can be referenced later. E.g. think how the first trades are inserted into the trade drop pod.

Right now; my animation blocks can detect structures; and know their bounds. So tomorrow I'll be add some simple graphics for it client side, and then saving them out to file using NBTIO (which I found exists, which is nice). 

Once that is all done; I can move onto the actual keyframe work. Which I've had some ideas of copying how create schmatics show the ghost structures and how you can move them. However, it will be different as you'll be able to see multiple at once. I will probably have to add a "key frame controller" item which you can use to go forwards or backwards between frames, and mark new key frames. Which sounds complex; but I aim to keep it simple. If that means only having "one direction" e.g. only go forward between frames.

I'll play with it when I get there to figure out the best UX to workflow ratio

o7 Also why no crouch in creative mode flying anymore SMH

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196540679749373963) 
> Currently stuck. It seems from what I can gather is that me updating the entity data on the server isn't being reped onto the clients

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196543909304750081) 
> [tulkinrb](https://discord.com/channels/890222432605057044/1194730593846558810/1196543909304750081) didn't solve this already?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196543974081564702) 
> I remember him talking about something along those lines

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196544171410980864) 
> Ah; instead of entity data, seems to be using persistent data?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196544284904661132) 
> I was about to go down the scary route of sending network packets to the client

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196552665056956588) 
> Persistent data doesn't seem to be working for me either

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196553036588404818) 
> Oh; it seems Tulk did use network events

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196554535976570942) 
> Tulk was a brave lad

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196554610052173885) 
> https://tenor.com/view/dwight-schrute-the-office-salute-yes-sir-rip-gif-16721894

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196554666322956318) 
>

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196554674724147321) 
> He was the best of us... Taken too soon <:cry:1184476833090113606>

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/335010807555293184/b4df6f36c4e6273f4e5a057dd0769fe7.webp" width="15" height="15" center=true/> **tulkinrb** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196562122205831198) 
> Meow

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/335010807555293184/b4df6f36c4e6273f4e5a057dd0769fe7.webp" width="15" height="15" center=true/> **tulkinrb** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196562189214040135) 
> Lol did I?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196562414691422258) 
> ```js
NetworkEvents.dataReceived("landing_spot_preview_stage_change", event => {
    current_preview_state = event.getData().getString("state");
});
```#

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196562690655649863) 
> persisent data didn't seem to work for me for server => client

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/335010807555293184/b4df6f36c4e6273f4e5a057dd0769fe7.webp" width="15" height="15" center=true/> **tulkinrb** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196562852572573878) 
> Wait for serverwide data as well?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196563076351262721) 
> ```js
entity.tick(20, 0, (blockEntity) => {
        if (blockEntity.level.isClientSide()) {
          let {x: bx, y: by, z:bz} = blockEntity.getBlockPos()
          let min = blockEntity.persistentData.getIntArray("min")
          let max = blockEntity.persistentData.getIntArray("max")
          console.log(min.length + ":" + max.length) // Doesn't get anything
          if (min.length >= 3 && max.length >= 3) {
            console.log("hit")
            let [minX, minY, minZ] = min
            let [maxX, maxY, maxZ] = max
            minX += bx, maxX += bx
            minY += by, maxY += by
            minZ += bz, maxZ += bz
            CreateClient.OUTLINER.showAABB(blockEntity, AABB.of(minX, minY, minZ, maxX, maxY, maxZ), 20)
              .colored(7580121)
              .withFaceTexture(SpecialTextures.THIN_CHECKERED, SpecialTextures.HIGHLIGHT_CHECKERED)
              .disableLineNormals(SelectionTexture, SelectionTexture)
              .lineWidth(0.06)
          }
        }
      })
```

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/335010807555293184/b4df6f36c4e6273f4e5a057dd0769fe7.webp" width="15" height="15" center=true/> **tulkinrb** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196563472956264499) 
> Yeah that's very weird, client-server things are kinda broken lol

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196564010309529703) 
> the worse part is this function is within the startup scripts. So restart the game each change to find it still don't work 🃏

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/335010807555293184/b4df6f36c4e6273f4e5a057dd0769fe7.webp" width="15" height="15" center=true/> **tulkinrb** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196564085282717786) 
> Oh no

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/335010807555293184/b4df6f36c4e6273f4e5a057dd0769fe7.webp" width="15" height="15" center=true/> **tulkinrb** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196564122033213521) 
> Startup scripts are a pain

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196570847578693642) 
> https://docs.minecraftforge.net/en/1.19.x/blockentities/#synchronizing-the-data-to-the-client

Honestly; sometimes the docs have the perfect thing you're looking for

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196574378226368663) 
> *The moment when you realise a block entity has 3-4 different 'compoundTags', so this whole time I've been accessing difference ones constantly*

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196576559826469016) 
> Found 5 different ways to give blocks data so far

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196590542184775751) 
> After what feels like forever

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196590586124316792) 
> in the block entity definition.

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196590693913727036) 
> I had to add
`entity.enableSync()`
and then after I update `entity.data` call `entity.sync()`

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196711979079962624) 
> maybe you can put the login in startup script but outside startup events?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196712183149633576) 
> on server side?

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196753065123008542) 
> Both; since the code is ran in the startup

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196753190746599466) 
> then whenever I want to send the data from the server to the client; then I'd call entity.sync

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196753972145430609) 
> This is probably the most compilated block we have added so far lmao 
```js
const UtilsJS = Java.loadClass('dev.latvian.mods.kubejs.util.UtilsJS')
const CreateClient = Java.loadClass('com.simibubi.create.CreateClient')
const SpecialTextures = Java.loadClass('com.simibubi.create.AllSpecialTextures')

StartupEvents.registry("block", (event) => {
  event.create("ptdye:animation_block")
    .translationKey("block.ptdye.animation_block")
    .tag("forge:immovable")
    .soundType(SoundType.STONE)
    .material("stone")
    .hardness(-1)
    .item((blockItem) => {
      blockItem.rarity("epic")
      blockItem.tooltip(Text.translate("block.ptdye.animation_block.tooltip").gold())
      blockItem.group(null) // remove from creative tabs
    })
    .blockEntity((entity) => {
      entity.enableSync()
      entity.initialData({
        name: "",
        min: [],
        max: [],
      })
      entity.tick(20, 0, (blockEntity) => {
        if (blockEntity.level.isClientSide()) {
          let {x: bx, y: by, z:bz} = blockEntity.getBlockPos()
          let min = blockEntity.data.getIntArray("min")
          let max = blockEntity.data.getIntArray("max")
          if (min.length >= 3 && max.length >= 3) {
            let [minX, minY, minZ] = min
            let [maxX, maxY, maxZ] = max
            minX += bx, maxX += bx
            minY += by, maxY += by
            minZ += bz, maxZ += bz
            CreateClient.OUTLINER.showAABB(blockEntity, AABB.of(maxX, maxY, maxZ, minX, minY, minZ), 20)
              .colored(7580121)
              .withFaceTexture(SpecialTextures.CHECKERED)
              .disableLineNormals()
              .lineWidth(0.06)
          }
        }
      })
    })
    .property(BlockProperties.HORIZONTAL_FACING)
    .property(BlockProperties.TRIGGERED)
    .placementState((state) => {
      state.setValue(BlockProperties.HORIZONTAL_FACING, Direction.north)
      state.setValue(BlockProperties.TRIGGERED, UtilsJS.cast(false))
    })
})
```

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196754257760751617) 
> So it was a lot of looking through forge docs, then looking through kubejs sauce to see if they've implemented the sync thing. Then working backwards to figure out how to use/enable it

**Comment by Yarden-zamir** [comment]
after you load an external class probe again, it should provide completions

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1196824781861965905) 
> It's because I was using `level.getBlockEntity` which returns a `BlockEntity` - but I needed `BlockEntityJS` which extends/inherits from `BlockEntity` and JavaScript does no casting. Sooo. I had to hope I could access `data` even though it doesn't have that field.
```js
const blockEntity = event.level.getBlockEntity(blockPos)
if (!(blockEntity instanceof BlockEntityJS))
  return
```
I got no idea if it will work; but I hope this will at least try to catch anything that isn't `BlockEntityJS`

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1198590463490408528) 
> ![animation_block.mp4](https://cdn.discordapp.com/attachments/1194730593846558810/1198590462550868088/animation_block.mp4?ex=65bf755b&is=65ad005b&hm=2d5b8d5baf426dfd4336e848c6d05e7f31c6a47bb57f75bc3aa74eb5e9ba13e5&)

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1198609577776795760) 
> oh I'm exited now, this looks epic

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199190923280662548) 
> Me talking to myself to get a little feedback.

I've decided to write a java function for the actual creation of the contraption nbt than have it in javascript.

I'm not sure how javascript -> Java works. So I'm assuming it is some what like lua -> C[++]. Where after each lua call to C it has to then jump back to lua and that can cause some cache misses, and adds extra cycles that don't need to be.

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199191044294705222) 
> and yes; I woke up at 1-2am to start this function and now I shall try to sleep again

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199652973245104138) 
> ```js
/**
 * Get `ptdye:animation_anchor_set` with position set
 * @param {BlockPos} blockPos 
 * @returns {Internal.ItemStack}
 */
function animation_getAnchorAt(blockPos) {
  const {x, y, z} = blockPos;
  const pos = [UtilsJS.parseInt(x, 0), UtilsJS.parseInt(y, 0), Utils.parseInt(z, 0)]
  let item = Item.withNBT("ptdye:animation_anchor_set", {
    target: NBT.intArrayTag(pos)
  })
  item = item.withName(Text.translate("item.ptdye.animation_anchor_set", pos.join(' ,'))) // this is italitic
  return item
}
```

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199653123921289316) 
> I could set the nbt easily tho since I'm already doing it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199656869363466250) 
> might be the text translate is already one

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199657841389207582) 
> Text.translate returns a text component

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199658149439885443) 
> Ooo it works

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199772195812020438) 
> Update time!
I've finished the first system. It seems robust and feature rich for what is required for it. Right now it saves the contraptions to a directory.

Next I'm going to try and define what I have to do next and how to achieve it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199778491172925450) 
> https://github.com/game-design-driven/ptdye-plus/pull/13

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199778855527923833) 
> [yarden.zamir](https://discord.com/channels/890222432605057044/1194730593846558810/1199778855527923833) I know how busy you are with testing your typing speed - but it would be handy to have a message in the chat when the issue is mentioned in another so I don't have to post a link in discord too

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199780745565196409) 
> Awesome! great job

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199781832246427728) 
> lol, done with that now, only needed to top past yarden, he was taunting me. Yes I can add that, my plan is to just support prs as if they are regular issues and they will have their own dev thread. Have a few more things I wanted to add and change but been procrastinating on that as writing a discord bot is not as exciting as working on the pack lol

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199782260648456363) 
> Yee yee; no pressure. work on what you want to

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199783434030166076) 
> Just remember I believe the bot would come under community so it is on the low priority

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199783478489784361) 
> So don't spend your life on it

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/127395969997209600/ed6d18deecd1592788b9747ed256e3b0.webp" width="15" height="15" center=true/> **yarden.zamir** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199785306115821708) 
> oh that reminds me, should move the bot to the org too

**Comment by Discord-Github-Bridge** [comment]
[<img src="https://cdn.discordapp.com/avatars/162262914898722817/b183775fb56dab06743d29fe1087f6b1.webp" width="15" height="15" center=true/> **engineersmith_4628** on Discord says](https://discord.com/channels/890222432605057044/1194730593846558810/1199792479533281330) 
> https://github.com/game-design-driven/Create-Prepare-to-Dye/pull/544
